packages/capabilities/src/access.js

```diff
@@ -108,3 +108,86 @@ export const claim = base.derive({
   }),
   derives: equalWith,
 })
+
+// https://github.com/web3-storage/specs/blob/main/w3-access.md#accessdelegate
+export const delegate = base.derive({
+  to: capability({
+    can: 'access/delegate',
+    /**
+     * Field MUST be a space DID with a storage provider. Delegation will be stored just like any other DAG stored using store/add capability.
+     *
+     * @see https://github.com/web3-storage/specs/blob/main/w3-access.md#delegate-with
+     */
+    with: DID.match({ method: 'key' }),
+    nb: {
+      // keys SHOULD be CIDs, but we won't require it in the schema
+      delegations: Schema.dictionary({
+        value: Schema.Link.match(),
+      }),
+    },
+    derives: (claim, proof) => {
+      return (
+        fail(equalWith(claim, proof)) ||
+        fail(subsetsNbDelegations(claim, proof)) ||
+        true
+      )
+    },
+  }),
+  derives: (claim, proof) => {
+    // no need to check claim.nb.delegations is subset of proof
+    // because the proofs types here never include constraints on the nb.delegations set
+    return fail(equalWith(claim, proof)) || true
+  },
+})
+
+/**
+ * Parsed Capability for access/delegate
+ *
+ * @template {Types.Ability} A
+ * @template {Types.URI} R
+ * @typedef {Types.ParsedCapability<A, R, { delegations?: Types.Failure | Schema.Dictionary<string, Types.Link<unknown, number, number, 0 | 1>> }>} ParsedAccessDelegate
```

Comment:
I use this below on lines 159, 160. @Gozala lmk if there is some better way of inferring this type. It's possible I can't just infer it from `delegate` because it's also used in the definition of `delegate` (derives) so there is a circularity?

Reply:
`delegations` should not be `Failure`,  ucanto will not create parsed capability like that. It will either parse successfully or just fail, it will never partially fail.

Reply:
```suggestion
 * @typedef {Types.ParsedCapability<A, R, { delegations?: Schema.Dictionary<string, Types.Link<unknown, number, number, 0 | 1>> }>} ParsedAccessDelegate
```

Does this not work ?

Reply:
@Gozala and I met together, and realized I needed to [add this type assertion](https://github.com/web3-storage/w3protocol/pull/420/commits/4f1f2931cecff1d1d1d29e889c4fdfb63ff3b327#diff-e434cc6c1a699df311a0b2faed199a2ff6b6b291d30f95e20b2ea5abfa7da3d9R125)

Reply:
I proposed this in ucanto to make the type assertion unnecessary https://github.com/web3-storage/ucanto/pull/215